modules/installation-osp-kuryr-settings.adoc

```diff
@@ -0,0 +1,175 @@
+in order to speed up creation of the new pods. Without port pools Kuryr would
+need to request a Neutron port creation each time a new Pod is spawned and
+port deletion when Pod gets removed. Those are relatively expensive operations,
+and the idea of port pools is to do them asynchronously whenever possible.
+
+The feature makes Kuryr keep a _pool_ of Neutron ports ready to be connected to
+Pod on creation. Neutron ports used by Kuryr are created in a subnet and
+attached as subports to a trunk port - the OpenShift node port. As Kuryr
+keeps each namespace in a separate subnet, it effectively means that a
+separate port pool is kept for each namespace-worker pair.
+
+Options below are available to adjust port pools behavior:
+
+* `enablePortPoolsPrepopulation` allows you to enable pool prepopulation,
+  which will make Kuryr add ports to the pool when it is created, i.e. when
+  new host is added or new namespace created. It is disabled by default,
+  meaning that pool is populated with ports when first Pod matching the pool
+  gets created.
+* `poolMinPorts` is the minimum number of free ports that should be kept in
+  the pool. If the number of free ports is lower new Neutron ports get created
+  for the pool. Defaults to 1.
+* `poolMaxPorts` is the maximum number of free ports kept in the pool. When
+  the number is higher additional Neutron ports will get deleted. `0` disables
+  that upper bound and is the default setting. If your OpenStack port quota is
+  low, or you have a limited number of IP addresses on the pod network you
+  should consider setting this option to some value to make sure ports are
+  deleted when not needed.
+* `poolBatchPorts` controls the number of Neutron ports getting created at
+  once when there's need. The default is `3`.
+
+.Prerequisites
+
+* Create the `install-config.yaml` file and complete any modifications to it.
+
+.Procedure
+
+. Use the following command to create manifests:
++
+[source,terminal]
+----
+$ ./openshift-install create manifests --dir=<installation_directory> <1>
+----
+<1> For `<installation_directory>`, specify the name of the directory that
+contains the `install-config.yaml` file for your cluster.
+
+. Create a file that is named `cluster-network-03-config.yml` in the
+`<installation_directory>/manifests/` directory:
++
+[source,terminal]
+----
+$ touch <installation_directory>/manifests/cluster-network-03-config.yml <1>
+----
+<1> For `<installation_directory>`, specify the directory name that contains the
+`manifests/` directory for your cluster.
++
+After creating the file, several network configuration files are in the
+`manifests/` directory, as shown:
++
+[source,terminal]
+----
+$ ls <installation_directory>/manifests/cluster-network-*
+----
++
+.Example output
+[source,terminal]
+----
+cluster-network-01-crd.yml
+cluster-network-02-config.yml
+cluster-network-03-config.yml
+----
+
+. Open the `cluster-network-03-config.yml` file in an editor and enter a CR that
```

Comment:
Will need to work through this a bit after other parts of the PR are resolved.

Reply:
Just a note - this is basically a customized copy of this module: https://github.com/openshift/openshift-docs/blob/18dcefa1bb6178d6aeaf806af0a7f1b9618aebb5/modules/nw-aws-nlb-new-cluster.adoc.